blockchain/chain_makers.go

```diff
@@ -177,8 +178,15 @@ func GenerateChain(config *params.ChainConfig, parent *types.Block, engine conse
 			TriesInMemory:       DefaultTriesInMemory,
 			TrieNodeCacheConfig: statedb.GetEmptyTrieNodeCacheConfig(),
 		}
+
+		prevConfig := fork.GetHardForkBlockNumberConfig()
 		blockchain, _ := NewBlockChain(db, cacheConfig, config, engine, vm.Config{})
-		defer blockchain.Stop()
+		defer func() {
+			blockchain.Stop()
+			if prevConfig != nil {
+				fork.SetHardForkBlockNumberConfig(prevConfig)
```

Comment:
Do we need this? If `NewBlockchain` is called later, the config will be set with a new value.

Reply:
I searched how GenerateChain is used. It is used  to generate test blocks and receipts. The blockchain created in here is not used outside. 

Also, in many times, generateChain is called when already `NewBlockchain` is running. It means hardForkBlockNumberConfig is already set. So I wanted to restore original value, so i put these lines.

Reply:
I'll reconsider it about this, thanks!